### PR TITLE
I18n support for hotkeys

### PR DESCRIPTION
**Pull request by wilzbach**
A suggestion for adding internationalization to the generated hotkeys overview (aka cheatsheet).
It's a bit tricky because this is generated from the JS side and we need to provide and set the keys from JS.

### CODE REVIEW

File: views/tour.dt [hunk 1]
```diff
@@ -9,6 +9,7 @@ block head
 	link(rel="stylesheet", href="/static/css/ansi.css")
 
 block content
+	script var translations = #{translations.toJson};
```
Comment (wilzbach) [inline views/tour.dt]: I'm not a huge fan of this. I guess ideally we would have a separate route which returns a json file per language. The returned translation file could be nicely cached.
However, 
- currently it's not a huge overhead
- I plan on avoiding the page reloads and making them as pure Ajax requests instead -> the translation file would only be requested once.

**Comment by dmi7ry** [comment]
Am I understood right that should add texts [this way](https://github.com/dlang-tour/russian/blob/master/index.yml)?

**Comment by wilzbach** [comment]
> Am I understood right that should add texts this way?

Yes, but it hasn't been merged yet. I wanted to wait for feedback, but seeing that you are very much in need of this I will hit merge and it should be deployed soon :)
Btw is it interesting for you to
- localize tour.dlang.org automatically based on the user language header?
- localize [run.dlang.io](https://run.dlang.io)?

**Comment by dmi7ry** [comment]
Russian version doesn't work now, source code finished here:

    <script>var translations = {"hotkeys.run_source_code":"

> localize tour.dlang.org automatically based on the user language header?

Personally I prefer hint like "That page available on your language" when open first time and small link at 
top of the page for fast (one-click) switch of language (one-click for based on the user language header and dropdown menu for all available languages).

> run.dlang.io

If I understand right what this is, then no (anyway each programmer should know English)

**Comment by wilzbach** [comment]
> Russian version doesn't work now, source code finished here:

That looks weird - the stack trace I receive points more to a Vibe.d problem:

```
#0  0x00007ffff692f6df in __memmove_avx_unaligned_erms () from /usr/lib/libc.so.6
#1  0x00005555561bca06 in _d_arrayappendT ()
#2  0x00005555561b0882 in object.Throwable.toString().__lambda1!(const(char[])).__lambda1(const(char[]))
    ()
#3  0x00005555561b0a1b in object.Throwable.toString(scope void(const(char[])) delegate) const.__foreachbody2(ref const(char[])) ()
#4  0x00005555561b22de in core.runtime.defaultTraceHandler(void*).DefaultTraceInfo.opApply(scope int(ref const(char[])) delegate) const.__lambda2!(ulong).__lambda2(ref ulong, ref const(char[])) ()
#5  0x00005555561bf8c9 in rt.backtrace.dwarf.traceHandlerOpApplyImpl(const(void*[]), scope int(ref ulong, ref const(char[])) delegate) ()
#6  0x00005555561b2321 in core.runtime.defaultTraceHandler(void*).DefaultTraceInfo.opApply(scope int(ref ulong, ref const(char[])) delegate) const ()
#7  0x00005555561b22ba in core.runtime.defaultTraceHandler(void*).DefaultTraceInfo.opApply(scope int(ref const(char[])) delegate) const ()
#8  0x00005555561b0994 in object.Throwable.toString(scope void(const(char[])) delegate) const ()
#9  0x00005555561b0846 in object.Throwable.toString() ()
#10 0x000055555606394e in vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool, scope std.experimental.allocator.IAllocator).__lambda22() (this=0x7ffff48819d0)
    at ../../vibe/vibe.d-repo/http/vibe/http/server.d:2167
#11 0x0000555556062f7b in vibe.http.server.handleRequest(vibe.core.stream.Stream, vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo, ref vibe.http.server.HTTPServerSettings, ref bool, scope std.experimental.allocator.IAllocator) (request_allocator=0x7ffff4881a10, keep_alive=@0x7ffff4881b50: true, 
    settings=@0x7ffff4881b48: 0x7ffff5a4bc00, listen_info=0x7ffff7f30280, 
    tcp_connection=0x5555564f9880, http_stream=0x5555564f9880)
    at ../../vibe/vibe.d-repo/http/vibe/http/server.d:2167
#12 0x000055555606165d in vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo).__lambda4() (this=0x7ffff4881b70)
    at ../../vibe/vibe.d-repo/http/vibe/http/server.d:1919
#13 0x00005555560614be in vibe.http.server.handleHTTPConnection(vibe.core.net.TCPConnection, vibe.http.server.HTTPListenInfo) (listen_info=0x7ffff7f30280, connection=0x5555564f9880)
    at ../../vibe/vibe.d-repo/http/vibe/http/server.d:1920
#14 0x0000555556060ccd in vibe.http.server.listenHTTPPlain(vibe.http.server.HTTPServerSettings, vibe.http.server.HTTPServerContext).doListen(vibe.http.server.HTTPListenInfo, vibe.http.server.HTTPServerContext, bool, bool).__lambda6(vibe.core.net.TCPConnection) (this=0x7ffff5c0be00, conn=0x5555564f9880)
    at ../../vibe/vibe.d-repo/http/vibe/http/server.d:1785
#15 0x0000555556182464 in vibe.core.drivers.libevent2_tcp.ClientTask.execute() (this=...)
    at ../../vibe/vibe.d-repo/core/vibe/core/drivers/libevent2_tcp.d:610
#16 0x0000555555e7b288 in vibe.core.core.makeTaskFuncInfo!(void() delegate).makeTaskFuncInfo(ref void() delegate).callDelegate(vibe.core.core.TaskFuncInfo*) (tfi=0x7ffff4881df0)
    at ../../vibe/vibe.d-repo/core/vibe/core/core.d-mixin-632:632
#17 0x0000555556115f3b in vibe.core.core.CoreTask.run() (this=0x7ffff5ad0a00)
    at ../../vibe/vibe.d-repo/core/vibe/core/core.d:1269
#18 0x000055555620d360 in core.thread.Fiber.run() ()
#19 0x000055555620d0bf in fiber_entryPoint ()
#20 0x0000000000000000 in ?? ()
```

I will have a look whether we can workaround it.

**Comment by wilzbach** [comment]
> I will have a look whether we can workaround it.

But unfortunately not today -> reverting: https://github.com/dlang-tour/core/pull/613
Sorry.

**Pull request merged**